index.js

```diff
@@ -39,18 +40,37 @@ async function getTds () {
  * @param {string} platform - platform to write
  * @param {object} config - the object to write
  */
-function writeConfigToDisk (platform, config, v1Config) {
+function writeConfigToDisk (platform, config) {
     let configName = platform
     if (platform.includes(BROWSERS_SUBDIR)) {
         configName = platform.replace(BROWSERS_SUBDIR, 'extension-')
     }
 
-    // Now that the features have been finalized, write out their hashes
+    // Convert config to backwards compatible versions
+    let compatConfig = config
+    const configsToWrite = []
+    for (const version of ['v2', 'v1']) {
+        if (!compatibility.compatFunctions[version]) {
+            continue
```

Comment:
This seems unexpected, should we throw here to prevent broken config parsing and uploading?